lib/reporters/tap.js

```diff
@@ -1,14 +1,44 @@
 'use strict';
 const format = require('util').format;
+const indentString = require('indent-string');
 const stripAnsi = require('strip-ansi');
+const extractStack = require('../extract-stack');
 
 // Parses stack trace and extracts original function name, file name and line
-function getSourceFromStack(stack, index) {
-	return stack
-		.split('\n')
-		.slice(index, index + 1)
-		.join('')
-		.replace(/^\s+ /, '');
+function getSourceFromStack(stack) {
+	return extractStack(stack).split('\n')[0];
+}
+
+function yamlBlockPropertyValue(value) {
+	const string = String(value);
+	return string.includes('\n') ?
+		`|\n${indentString(string, 6)}` :
+		string;
+}
+
+function yamlBlock(error, includeMessage) {
+	const output = [];
+
+	if (error.name) {
+		output.push(`    name: ${yamlBlockPropertyValue(error.name)}`);
+	}
+	if (includeMessage && error.message) {
+		output.push(`    message: ${yamlBlockPropertyValue(error.message)}`);
```

Comment:
I think we need to handle quotes.

For example:

```
message: 'hello' this is not valid YAML
```

Wouldn't it be better to just use a real YAML stringifier? Like [`js-yaml`](https://github.com/nodeca/js-yaml)